snooty/test_parser.py

```diff
@@ -3730,3 +3730,133 @@ def test_duplicate_directive_option() -> None:
     )
     page.finish(diagnostics)
     assert [type(d) for d in diagnostics] == [DocUtilsParseError]
+
+
+def test_inconsistent_levels() -> None:
+    """DOP-3969"""
+    path = FileId("test.rst")
+    project_config = ProjectConfig(ROOT_PATH, "")
+    parser = rstparser.Parser(project_config, JSONVisitor)
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+Query Result Format
```

Comment:
Can you add some comments to these describing why they're wonky or not wonky so it's clearer to scan?

Reply:
Done! ✍🏻